docs/user_needs/zephyr_rtos_user.sdoc

```diff
@@ -0,0 +1,87 @@
+[DOCUMENT]
+TITLE: Zephyr RTOS User
+REQ_PREFIX: ZEP-
+
+[GRAMMAR]
+IMPORT_FROM_FILE: user_needs.sgra
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[USER_NEED]
+UID: ZEP-127
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: OSS License Allows Statically Linked Private Extensions
+STATEMENT: >>>
+As a Product Manufacturer of resource-constrained devices,
+
+I need open-source software to be licensed such that the proprietary code statically linked with the open source software can be kept private.
+<<<
+
+[USER_NEED]
+STATUS: Draft
+TYPE: User Need
+USER: Zephyr RTOS User
+TITLE: Easily Switch Application to different MCU Architecture
+STATEMENT: >>>
+As a Zephyr RTOS user,
+
+I want to be able to easily switch my application to a different MCU architecture (x86, ARM Cortex-M/A, RISCV etc.).
+<<<
+
+[USER_NEED]
+UID: ZEP-128
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: Development Systems Options for Windows, Linux
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need embedded software development system options for Windows and Linux host operating systems.
+<<<
+
+[USER_NEED]
+UID: ZEP-129
+STATUS: Draft
+TYPE: User Need
+USER: Product Manufacturer
+TITLE: RTOS with Preemptive Scheduling
+STATEMENT: >>>
+As a Product Manufacturer,
+
+I need an RTOS that provides preemptive scheduling of threads,
```

Comment:
There are two different things hiding here: 

1) a more general one that we need to manage threads/tasks for parallel execution of activities.
2) preemptive scheduling can go right after the first, and the current `so that ...` part applies to this second one.

Reply:
Agreed. We may want to break these into two distinct needs.

Reply:
Hmm ... can an OS support preemptive scheduling without it being an RTOS?

Reply:
For the first point, a COMMENT could maintain a ladder: `baremetal` -> `RTOS` -> `Linux`. RTOS does parallel task execution and management but is still very small and simple compared to Linux.

Reply:
1) "Parallel" execution is rather "Concurrent".
2) We need a glossary definition of "RTOS".
3) We do need to separate preemptive scheduling from RTOS. Extract the first part of the statement from the second. Distill the second part further if needed.